### PR TITLE
Move drag related functions to the DragHandle component

### PR DESCRIPTION
**Pull request by amanmahajan7**

### CODE REVIEW

File: src/DragHandle.tsx [hunk 1]
```diff
@@ -19,6 +23,108 @@ const cellDragHandle = css`
...
+  columns,
+  selectedPosition,
+  draggedOverRowIdx,
+  onRowsChange,
+  onFill,
+  setDragging,
+  setDraggedOverRowIdx
+}: Props<R, SR>) {
+  const latestDraggedOverRowIdx = useRef(draggedOverRowIdx);
+  useEffect(() => {
```
Comment (amanmahajan7) [inline src/DragHandle.tsx:45]: I think `useEffect` is fine here but we can pass the `latestDraggedOverRowIdx` ref instead if you think it is more safe

Comment (nstepien) [reply src/DragHandle.tsx:46]: Could just do
```ts
function handleDragEnd() {
  setDraggedOverRowIdx((draggedOverRowIdx) => {
    // ...
    return undefined;
  });
}
```

File: src/DragHandle.tsx [hunk 2]
```diff
@@ -19,6 +22,106 @@ const cellDragHandle = css`
 
 const cellDragHandleClassname = `rdg-cell-drag-handle ${cellDragHandle}`;
 
-export default function DragHandle(props: React.HTMLAttributes<HTMLDivElement>) {
-  return <div className={cellDragHandleClassname} {...props} />;
+interface Props<R, SR> extends Pick<DataGridProps<R, SR>, 'rows' | 'onRowsChange'> {
+  columns: readonly CalculatedColumn<R, SR>[];
+  selectedPosition: SelectCellState | EditCellState<R>;
```
Comment (nstepien) [inline src/DragHandle.tsx:27]: I don't think it's possible
```suggestion
  selectedPosition: SelectCellState;
```

File: src/DragHandle.tsx [hunk 3]
```diff
@@ -19,6 +22,106 @@ const cellDragHandle = css`
...
+    function onMouseUp() {
+      window.removeEventListener('mouseover', onMouseOver);
+      window.removeEventListener('mouseup', onMouseUp);
+      setDragging(false);
+      handleDragEnd();
+    }
+  }
+
+  function handleDragEnd() {
+    if (!onRowsChange) return;
```
Comment (nstepien) [inline src/DragHandle.tsx:64]: Should disable the feature if `onRowsChange` isn't defined

Comment (amanmahajan7) [reply src/DragHandle.tsx:64]: I think we should allow it if `onFill != null`. It is like editor, we do not block editor if `onRowsChange` is not specified.

Comment (nstepien) [reply src/DragHandle.tsx:64]: `onFill` cannot be called when `onRowsChange` is nullish though right now, should remove the return guards.

**Review by nstepien — APPROVED** [review]

**Pull request merged**